### PR TITLE
[Merged by Bors] - feat: `Nat.lcm_mul_left` and `Nat.lcm_mul_right`

### PR DESCRIPTION
**Pull request by amellendijk**
Add two lemmas about `Nat.lcm`. These results mirror [Nat.gcd_mul_left](https://leanprover-community.github.io/mathlib4_docs/Std/Data/Nat/Gcd.html#Nat.gcd_mul_left) present in Std.

---

Note we do have the more general [gcd_mul_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GCDMonoid/Basic.html#gcd_mul_left), which includes a `normalize` that  becomes trivial in the `Nat` case.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/GCD/Basic.lean [hunk 1]
```diff
@@ -103,6 +103,19 @@ theorem lcm_pos {m n : ℕ} : 0 < m → 0 < n → 0 < m.lcm n := by
...
 #align nat.lcm_pos Nat.lcm_pos
 
+theorem lcm_mul_left {m n k : ℕ} : (m * n).lcm (m * k) = m * n.lcm k := by
+  unfold lcm
+  by_cases hm : m = 0
+  · simp_rw [hm, zero_mul]
+  rw [gcd_mul_left, ←Nat.div_div_eq_div_mul, mul_assoc, Nat.mul_div_cancel_left
+    _ (Nat.pos_of_ne_zero hm), ←Nat.mul_div_assoc, Nat.mul_left_comm]
+  calc _ ∣ n := gcd_dvd_left ..
+       _ ∣ _ := Nat.dvd_mul_right ..
```
Comment (tb65536) [inline Mathlib/Data/Nat/GCD/Basic.lean:113]:
```suggestion
  apply dvd_antisymm
  · exact lcm_dvd (mul_dvd_mul_left m (dvd_lcm_left n k)) (mul_dvd_mul_left m (dvd_lcm_right n k))
  · have h : m ∣ lcm (m * n) (m * k) := (dvd_mul_right m n).trans (dvd_lcm_left (m * n) (m * k))
    rw [←dvd_div_iff h, lcm_dvd_iff, dvd_div_iff h, dvd_div_iff h, ←lcm_dvd_iff]
```

**Comment by alexjbest** [comment]
How about adding `Int.lcm_mul_right` (/right) here too?
I suppose they should also all be simp lemmas to match the non-namespaced versions?
It does make me question whether we should be using `Nat.lcm` for anything other than setting up and proving the general LCM theory and applying general lemmas.
What is your intended application?

File: Mathlib/Data/Nat/GCD/Basic.lean [hunk 2]
```diff
@@ -103,6 +103,19 @@ theorem lcm_pos {m n : ℕ} : 0 < m → 0 < n → 0 < m.lcm n := by
...
+
+theorem lcm_mul_right {m n k : ℕ} : (m * n).lcm (k * n) = m.lcm k * n := by
+  simp_rw [mul_comm]; rw [mul_comm]
+  exact lcm_mul_left
```
Comment (tb65536) [inline Mathlib/Data/Nat/GCD/Basic.lean:117]:
```suggestion
 rw [mul_comm, mul_comm k n, lcm_mul_left, mul_comm]
```

**Comment by FLDutchmann** [comment]
> How about adding `Int.lcm_mul_right` (/right) here too? I suppose they should also all be simp lemmas to match the non-namespaced versions? It does make me question whether we should be using `Nat.lcm` for anything other than setting up and proving the general LCM theory and applying general lemmas. What is your intended application?

I agree on the `Int` versions. As for `simp`, do we want to match the behaviour of [lcm_mul_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GCDMonoid/Basic.html#lcm_mul_left) or [Nat.gcd_mul_left](https://leanprover-community.github.io/mathlib4_docs/Std/Data/Nat/Gcd.html#Nat.gcd_mul_left)? The latter seems more consistent, and it would be more difficult to change that one to `simp` since it's in `Std`.

For my intended application it would definitely be possible to switch to `GCDMonoid.lcm`. The only place I use this lemma is for constructing a bijection between the set of pairs `(a, b)` with `a.lcm b = n` and the set of three-tuples `![x,y,z]` with product equal to `n` (specifically to show `(x*y).lcm (x*z) = x*y*z` for squarefree `n`)

That said I do think there is a case to be made for having parallel API for `Nat` (and `Int`). Note `Nat.lcm` is defined in `Std`. It's not inconceivable that someone would write a program using `Nat.lcm`, then later import `Mathlib` to prove something only to find out there's no API for it. I also think `gcd` and `lcm` are sufficiently elementary notions that programmers should be able to use them ergonomically without having to figure out what a `NormalizationMonoid` is. (And importantly the `Nat` case really *is* simpler, since you don't have to worry about normalisation)

**Comment by ocfnash** [comment]
It's a bit unfortunate that these changes will mean that [Nat.gcd_mul_right](https://leanprover-community.github.io/mathlib4_docs/Std/Data/Nat/Gcd.html#Nat.gcd_mul_right) (and left) are in `Std` but these changes will be the corresponding lcm versions in Mathlib.

It's also a bit unfortunate that we will not have consistent `simp` attributes between `Std` and `Mathlib`.

Surely it's not so hard to contribute to `Std`; how about adding all this to `Std` instead and introducing `simp` consistency?

Even though this is my preference, I'm not against adding these to Mathlib. I'll let @alexjbest make the call.

bors d=alexjbest

**Comment by alexjbest** [comment]
I don't have a particular preference, and am not really sure where the diving line between mathlib and std is (in theory I guess std should only have very generally useful things for formalizing both maths and software, but also not grow too large, alas that's not really something measurable).
Does the fact that nobody needed these lemmas to be in std yet mean they don't belong there for now? I do think gcds tend to come up more than lcms, so the lack of symmetry wouldn't be too surprising.

The simp-ness is a bigger issue I guess. I would say that they should all be simp so there should be at least one pr to std here.

**Comment by alexjbest** [comment]
Let's just merge this and move it later if needed I think :)
bors r+

**Pull request closed without merging**